internal/plugin.go

```diff
@@ -1237,6 +1240,40 @@ func assertValidPlacement(
 		}
 	}
 
+	if len(placement.ClusterSelectors) > 0 && len(placement.ClusterSelector) > 0 {
+		return fmt.Errorf("cannot use both clusterSelectors and clusterSelector in placement config")
+	}
+
+	selectorsUsed := map[string]bool{
+		"legacy":       false,
+		"fullSelector": false,
+	}
+	for key := range placement.ClusterSelector {
```

Comment:
I know it's a bit wasteful resource wise, but I think it'd be cleaner if you just ran `p.generatePlacement` and didn't use the output and just checked if the error was nil. All the validation is happening in that method already.